pyccel/parser/semantic.py

```diff
@@ -830,11 +830,14 @@ def _visit_str(self, expr, **settings):
 
     def _visit_Slice(self, expr, **settings):
         args = list(expr.args)
-        if args[0] is not None:
+        if args[0]:
             args[0] = self._visit(args[0], **settings)
 
-        if args[1] is not None:
+        if args[1]:
             args[1] = self._visit(args[1], **settings)
+
+        if args[2]:
+            args[2] = self._visit(args[2], **settings)
```

Comment:
After your changes in the `Slice` class you should be able to replace:
`args[0]` -> `start`
`args[1]` -> `end`
`args[2]` -> `step`